### PR TITLE
Brute Force - External Login Password Spray - Playbook + Triggers

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11361

## Description
New playbook to respond to successful/non-successful external login password sprays. Designed to handle the following alerts:

- External Login Password Spray
- Successful External Login Password Spray
- External Login Password Spray on a Domain Controller
- External Login Password Spray Involving a Honey User
- Successful External Login Password Spray on a Domain Controller
- Successful External Login Password Spray on a sensitive server

External_Login_Password_Spray

### CODE REVIEW

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 1]
```diff
@@ -0,0 +1,1474 @@
+id: External Login Password Spray
+version: -1
+name: External Login Password Spray
+description: |-
+  This playbook investigates and responds to external login password sprays. It starts by enriching the external IP, allowing early containment. It then fetches event information to learn more about the attack - how it was executed and whether it succeeded or not.
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:5]: This playbook investigates and responds to external login password sprays. It starts by enriching the external IP, allowing early containment. It then fetches event information to learn more about the attack - how it was executed and whether it succeeded or notThis playbook is designed to investigate and respond to external login password sprays. It begins by enriching the external IP to enable early containment. Then, it retrieves event information to gain insights into the attack, including how it was carried out and whether it was successful.

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 2]
```diff
@@ -0,0 +1,1474 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "45":
+    id: "45"
+    taskid: 119f0216-cd78-4fbc-81d8-3fc9e075d9fa
+    type: regular
+    task:
+      id: 119f0216-cd78-4fbc-81d8-3fc9e075d9fa
+      version: -1
+      name: Check reputation of Source IP
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:573]: Check Source IP Reputation

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 3]
```diff
@@ -0,0 +1,1474 @@
...
+      id: a1924082-41b3-47f8-8766-d268b29a9658
+      version: -1
+      name: Is the IP malicious?
+      description: Checks if the external Source IP is malicious (DBotScore above 2).
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "15"
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:617]: change  the flow so the playbook does not pause on the manual task, let the playbook continue

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 4]
```diff
@@ -0,0 +1,1474 @@
...
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "65":
+    id: "65"
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:906]: Why are we not setting the password for users who connected via RDP in soft remediation?

**Review by AdiPeret — CHANGES_REQUESTED** [review]

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 5]
```diff
@@ -0,0 +1,1305 @@
+id: b35490b2-f13c-4bd4-8c37-e795dca1db73
+version: 213
+vcShouldKeepItemLegacyProdMachine: false
+name: External Login Password Spray
+description: This playbook is designed to investigate and respond to external login
+  password sprays. It begins by enriching the external IP to enable early containment.
+  Then, it retrieves event information to gain insights into the attack, including
+  how it was carried out and whether it was successful.
+tags:
```
Comment (umishkin) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:5]: Tags should be [Technique - Name], [Tactic - Name]

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md
[the remark this answers is not in the document]
Comment (umishkin) [reply Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md:1]: Please upload the most updated playbook pic before the review next time, it's important for me to review it also and I'm looking at it

[the remark this answers is not in the document]
Comment (umishkin) [reply Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md:1]: On Sunday let me show you F2F how I think we need to change the playbook visually

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 6]
```diff
@@ -0,0 +1,1305 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "28":
+    id: "28"
+    taskid: 43bc9013-8020-4de9-8208-394c78020139
+    type: regular
+    task:
+      id: 43bc9013-8020-4de9-8208-394c78020139
+      version: -1
+      name: Get user risk score of logged in users
```
Comment (umishkin) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:363]: We tested that on multiple users right?

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 7]
```diff
@@ -0,0 +1,1305 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "67":
+    id: "67"
+    taskid: 868e45e4-7156-42ff-8e69-98cdd60d2a07
+    type: regular
+    task:
+      id: 868e45e4-7156-42ff-8e69-98cdd60d2a07
+      version: -1
+      name: Terminate RDP Session
```
Comment (umishkin) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:894]: Same question about multiple users, did we checked that on multiple users?

Comment (idovandijk) [reply Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:894]: Yes, checked without users, with one user, with 2 users where one should be logged off, etc.

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 8]
```diff
@@ -0,0 +1,1281 @@
...
+    isautoswitchedtoquietmode: false
+  "10":
+    id: "10"
+    taskid: e48b4bac-f8c7-40ee-8622-12f9c59e446f
+    type: condition
+    task:
+      id: e48b4bac-f8c7-40ee-8622-12f9c59e446f
+      version: -1
+      name: Any risky users to remediate?
+      description: "Checks whether hard remediation should be performed. \n\nTo perform hard remediation, all of the following conditions have to be met:\n\n1. At least one user logged in successfully.\n2. At least of the above users has a High risk level (available with ITDR license).\n3. The interval analysis suggests automation was used OR the failed logon attempts exceeded the baseline for the host.\n4. The alert severity is Medium or above."
```
Comment (ShirleyDenkberg) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:48]:
```suggestion
      description: "Checks whether hard remediation should be performed. \n\nTo perform hard remediation, all of the following conditions have to be met:\n\n1. At least one user logged in successfully.\n2. At least one of the logged in users has a High risk level (available with ITDR license).\n3. The interval analysis suggests automation was used OR the failed logon attempts exceeded the baseline for the host.\n4. The alert severity is Medium or above."
```

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 9]
```diff
@@ -0,0 +1,1281 @@
...
+    isautoswitchedtoquietmode: false
+  "13":
+    id: "13"
+    taskid: 64b2ac11-b9d3-422d-8ec1-bc4e1851228b
+    type: regular
+    task:
+      id: 64b2ac11-b9d3-422d-8ec1-bc4e1851228b
+      version: -1
+      name: Save all usernames that logged in successfully
+      description: Saves the users that logged in successfully in a new context key.
```
Comment (ShirleyDenkberg) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:146]:
```suggestion
      description: Saves the users who logged in successfully in a new context key.
```

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml [hunk 10]
```diff
@@ -0,0 +1,1281 @@
...
+    isautoswitchedtoquietmode: false
+  "28":
+    id: "28"
+    taskid: 43bc9013-8020-4de9-8208-394c78020139
+    type: regular
+    task:
+      id: 43bc9013-8020-4de9-8208-394c78020139
+      version: -1
+      name: Get user risk score of logged in users
+      description: Gets the risk score of the users that successfully logged in.
```
Comment (ShirleyDenkberg) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml:364]:
```suggestion
      description: Gets the risk score of the users who successfully logged in.
```

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
+This playbook investigates and responds to external login password sprays. It starts by enriching the external IP, allowing early containment. It then fetches event information to learn more about the attack - how it was executed and whether it succeeded or not.
+For remediation, the playbook checks for RemoteInteractive (RDP) sessions to close, and allows the analyst to expire the password of any users that successfully logged in as part of the attack.
```
Comment (ShirleyDenkberg) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md:2]:
```suggestion
For remediation, the playbook checks for RemoteInteractive (RDP) sessions to close, and allows the analyst to expire the password of any users who successfully logged in as part of the attack.
```

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+## Dependencies
+
+This playbook uses the following sub-playbooks, integrations, and scripts.
+
+### Sub-playbooks
+
+* PAN-OS - Block IP
```
Comment (ShirleyDenkberg) [inline Packs/Core/Playbooks/playbook-External_Login_Password_Spray_README.md:10]:
```suggestion
PAN-OS - Block IP
```

File: Packs/Core/ReleaseNotes/3_0_70.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
+
+#### Playbooks
+
+##### New: External Login Password Spray
+
+- New: Added a new playbook for investigation and response to external login password spray attacks. The playbook checks the logon type, the outcome, the timestamps, the users, and the IP, to decide on the correct remediation plan, prompted by user approval.
```
Comment (ShirleyDenkberg) [inline Packs/Core/ReleaseNotes/3_0_70.md:6]:
```suggestion
New: Added a new playbook for investigation and response to external login password spray attacks. The playbook checks the logon type, the outcome, the timestamps, the users, and the IP, to decide on the correct remediation plan, prompted by user approval. 
```

File: Packs/Core/ReleaseNotes/3_0_70.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+##### New: External Login Password Spray
+
+- New: Added a new playbook for investigation and response to external login password spray attacks. The playbook checks the logon type, the outcome, the timestamps, the users, and the IP, to decide on the correct remediation plan, prompted by user approval. 
+
+#### Triggers Recommendations
+
+##### New: External Login Password Spray
+
+- New: This trigger is responsible for handling External Login Password Spray attacks<~XSIAM> (Available for Cortex XSIAM).</~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/Core/ReleaseNotes/3_0_70.md:12]:
```suggestion
New: This trigger is responsible for handling External Login Password Spray attacks<~XSIAM> (Available for Cortex XSIAM).</~XSIAM>
```

**Comment by ShirleyDenkberg** [comment]
@umishkin @AdiPeret Doc review completed.

File: Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml (hunk 8 above)
Comment (idovandijk) [reply Packs/Core/Playbooks/playbook-External_Login_Password_Spray.yml]: Good catch Shirley!

**Comment by idovandijk** [comment]
Updated with notes from reviews and technical writer suggestions. Changed status to: Waiting for `PAN-OS - Block IP`.

**Review by AdiPeret — APPROVED** [review]

**Pull request merged**